### PR TITLE
Fixed memory leaks in cache and database reload, closes getlantern/go…

### PR DESCRIPTION
**Pull request by oxtoacart**
…-geoserve#3 and getlantern/go-geoserve#4

This fixes the memory leaks.  While in there, I also switched to using golog like our newer software.

### CODE REVIEW

File: geoserve/geoserve.go [hunk 1]
```diff
@@ -4,25 +4,33 @@ import (
...
+
+	"github.com/getlantern/golog"
+	"github.com/golang/groupcache/lru"
+	geoip2 "github.com/oschwald/geoip2-golang"
 )
 
 const (
 	DB_URL = "https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz"
+
+	CacheSize = 50000
```
Comment (oxtoacart) [inline geoserve/geoserve.go]: We can tweak this to achieve the right balance between memory consumption and CPU usage.  Basically, the cache helps avoid a little computation to look up the geolocation information, and a lot of computation to generate the JSON representation of the result.5

**Comment by uaalto** [comment]
Nice, LGTM!
We are logging to Heroku, right?

**Comment by uaalto** [comment]
By the way, I don't have permissions in this repo to merge or assign myself to this PR. Can you give me access please?

**Comment by uaalto** [comment]
I have now :)

**Comment by oxtoacart** [comment]
@uaalto Thanks for reviewing!  The app is set up to log to Papertrail in Heroku, yes.  I just gave you (and the rest of the core team) access.

**Comment by oxtoacart** [comment]
I've deployed this to Heroku, seems to be working okay so far.

**Pull request merged**